ruby/2022/day2/lib/solver.rb

```diff
@@ -0,0 +1,71 @@
+class RockPaperScissorsGame
+
+  attr_accessor :input_data
+
+  def initialize(input_data)
+    @input_data = input_data
+  end
+
+  def self.from_file(filepath)
+    new(File.foreach(filepath, chomp: true).map { |line| line.split(/ /) })
+  end
+
+  def solve_part1
+    input_data.reduce(0) do |total_self_score, round|
+      total_self_score + Round.from_raw(round).self_score
+    end
+  end
+end
+
+class Round
+  SHAPE_POINTS = {
+    rock: 1,
+    paper: 2,
+    scissors: 3,
+  }
+  OUTCOME_POINTS = {
+    win: 6,
+    draw: 3,
+  }
+  OPPONENT_SHAPE_KEY = {
+    "A" => :rock,
+    "B" => :paper,
+    "C" => :scissors,
+  }
+  SELF_SHAPE_KEY = {
+    "X" => :rock,
+    "Y" => :paper,
+    "Z" => :scissors,
+  }
```

Comment:
I wanted to be able to have semantic and expressive code later on. Had to define some of the logic from the instructions like this in order to get it (I think?).